perceval/serialization/deserialize.py

```diff
@@ -365,20 +371,32 @@ def resolve(self):
         if min_photons_filter != VALUE_NOT_SET:
             experiment.min_detected_photons_filter(min_photons_filter)
 
-        # Needed before adding the heralds
-        for serial_comp in self._pb_e.components:
-            component = CircuitBuilder.deserialize(serial_comp, self._params)
-            experiment.add(serial_comp.starting_mode, component)
-
-        # TODO: Detector deserialization needed before component deserialization if there are feed-forward configurators
-        #  However, we can't add photonic components after adding a detector (PCVL-799)
+        detectors: list[IDetector | None] = [None] * experiment.m
+        injected_detectors: list[bool] = [False] * experiment.m
         for i, serial_detector in self._pb_e.detectors.items():
             t = serial_detector.WhichOneof('type')
             serial_sub_comp = getattr(serial_detector, t)
             # find the correct deserialization function and use it
             func = self.deserialize_detector_fn[t]
             detector = func(serial_sub_comp)
-            experiment.add(i, detector)
+            detectors[i] = detector
+
+        # Needed before adding the heralds
+        for serial_comp in self._pb_e.components:
+            component = CircuitBuilder.deserialize(serial_comp, self._params)
+            if isinstance(component, AFFConfigurator):
+                for i in range(serial_comp.starting_mode, serial_comp.starting_mode + component.m):
+                    if not injected_detectors[i]:
+                        experiment._components.append(((i,), detectors[i]))
+                        injected_detectors[i] = True
+                experiment._components.append((tuple(i for i in range(serial_comp.starting_mode, serial_comp.starting_mode + component.m)), component))
+            else:
+                experiment.add(serial_comp.starting_mode, component)
+
+        # Add detectors
+        for i, d in enumerate(detectors):
+            if d is not None:
```

Comment:
Shouldn't we check that `injected_detectors[i] == False` ? I don't see what prevents adding twice the same detector

Reply:
In an `Experiment`, `Detector` instances appear twice when they're on a feed-forwarded mode. We definitely want to add all the detectors to the detector list because it's this list that's used during the output states post-processing.